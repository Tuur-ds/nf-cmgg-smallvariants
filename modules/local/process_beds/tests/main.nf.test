nextflow_process {

    name "Test Process PROCESS_BEDS"
    script "../main.nf"
    process "PROCESS_BEDS"

    tag "modules"
    tag "modules_local"
    tag "PROCESS_BEDS"

    test("homo_sapiens - bed") {

        config "./nextflow.config"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.bed, checkIfExists:true),
                    file(params.split1, checkIfExists:true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("homo_sapiens - bed - stub") {

        options "-stub"
        config "./nextflow.config"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.bed, checkIfExists:true),
                    file(params.split1, checkIfExists:true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
