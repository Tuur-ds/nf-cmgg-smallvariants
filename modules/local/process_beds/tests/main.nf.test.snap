{
    "homo_sapiens - bed": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,ffd6f70b7fa85ffb46fb66a4cf9afb70"
                    ]
                ],
                "1": [
                    "versions.yml:md5,c6be2bd45c01066eacc08f419bce67d6"
                ],
                "bed": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,ffd6f70b7fa85ffb46fb66a4cf9afb70"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c6be2bd45c01066eacc08f419bce67d6"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-23T16:45:26.442970884"
    },
    "homo_sapiens - bed - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,c6be2bd45c01066eacc08f419bce67d6"
                ],
                "bed": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c6be2bd45c01066eacc08f419bce67d6"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-23T16:45:35.803097114"
    }
}